src/fontra/views/applicationsettings/panel-base.js

```diff
@@ -0,0 +1,57 @@
+import { UndoStack, reverseUndoRecord } from "../core/font-controller.js";
+import * as html from "../core/html-utils.js";
+import { commandKeyProperty } from "../core/utils.js";
+
+export class BaseInfoPanel {
+  constructor(applicationSettingsController, panelElement) {
+    this.applicationSettingsController = applicationSettingsController;
+    this.panelElement = panelElement;
+  }
+
+  visibilityChanged(onOff) {
+    this.visible = onOff;
+    if (onOff && !this.initialized) {
+      this.initializePanel();
+      this.initialized = true;
+    }
+  }
+
+  initializePanel() {
+    this.undoStack = new UndoStack();
+    this.setupUI();
+  }
+
+  handleKeyDown(event) {
+    if (event[commandKeyProperty]) {
+      if (event.key == "z") {
+        event.stopImmediatePropagation();
+        event.preventDefault();
+        this.doUndoRedo(event.shiftKey);
+      }
+    }
+  }
+
+  async doUndoRedo(isRedo) {
```

Comment:
Is undo actually hooked up?

Reply:
It's not hooked up. It was pretty much a copy of the fontinfo panel-base.js. I left it almost unchanged, because I was not sure if the plan is to implement such functionality or not. 
I removed it.